Assets/Features/Building/Scripts/Grid/Grid.cs

```diff
@@ -0,0 +1,391 @@
+                Debug.LogError("Tile \"" + tile.name + "\" not found in Atlas");
+                return false;
+            }
+
+            return Set(gridVector, Array.FindIndex(_atlas.Items, x => x.Tile == tileData.Tile));
+        }
+
+        /// <summary>
+        /// Sets a group of cells if the target cells are empty
+        /// </summary>
+        /// <param name="gridVectors">Positions on grid</param>
+        /// <param name="tiles">Tiles that are to be placed. They have to be set in the BuildableAtlas</param>
+        /// <param name="rotation">Rotation of the individual tiles (0 to 3 clockwise)</param>
+        /// <returns>True if setting was a success</returns>
+        public bool SetGroup(List<Vector3Int> gridVectors, List<Tile> tiles, int rotation)
+        {
+            //check if all the positions are available
+            foreach (var position in gridVectors)
+            {
+                if (Get(position) != -1)
+                    return false;
+            }
+
+            for (int i = 0; i < gridVectors.Count; i++)
+            {
+                //get tile from atlas
+                TileData tileData = _atlas.Items.FirstOrDefault(x => x.Tile == tiles[i]);
+
+                if (tileData == default)
+                {
+                    Debug.LogError("Tile \"" + tiles[i].name + "\" not found in Atlas");
+                    return false;
+                }
+
+                var cellData = _cells[gridVectors[i].x, gridVectors[i].y, gridVectors[i].z];
+
+                cellData.Tile = Array.FindIndex(_atlas.Items, x => x.Tile == tileData.Tile);
+                cellData.Rotation = rotation;
+
+                _cells[gridVectors[i].x, gridVectors[i].y, gridVectors[i].z] = cellData;
+            
+
+                _mapUpdated = true;
+            }
+
+            GameManager.Instance.TaskManager.BuilderTaskSystem.GetTaskSystem().AddTask(new BuildTask(gridVectors[0]));
+            _cellGroup.Add(gridVectors);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Removes a tile from the grid
+        /// </summary>
+        /// <param name="gridVector">Position to be removed</param>
+        /// <returns>True if remove was successful</returns>
+        public bool Remove(Vector3Int gridVector)
+        {
+            if (!OutOfBounds(gridVector) && Get(gridVector) != -1)
+            {
+                //checks if given tile is part of a linked group. If not only add the given position
+                var group = _cellGroup.FirstOrDefault(x => x.Any(j => j == gridVector));
+                if (group == default)
+                    group = new List<Vector3Int>() { gridVector };
+
+                //Remove from array
+                foreach (var item in group)
+                {
+                    _cells[item.x, item.y, item.z].Tile = -1;
+                    _cells[item.x, item.y, item.z].Rotation = 0;
+                    _cells[item.x, item.y, item.z].BuildPercentage = 0.4f;
```

Comment:
`Buildpercentage` is hardcoded?

Reply:
For the time being yes. Just so we have a quick visualisation. The build percentage is directly coupled to the opacity